odxtools/parameterinfo.py

```diff
@@ -1,105 +1,187 @@
+            of.write(f": enum; choices:\n")
             for scale in cm.internal_to_phys:
-                result += f"  '{str(scale.compu_const)}'\n"
+                val_str = ""
+                if scale.lower_limit is not None:
+                    val_str = f"({repr(scale.lower_limit.value)})"
+                of.write(f"  {repr(scale.compu_const)}{val_str}\n")
 
         elif isinstance(cm, IdenticalCompuMethod):
             bdt = dop.physical_type.base_data_type
             if bdt in (DataType.A_UTF8STRING, DataType.A_UNICODE2STRING, DataType.A_ASCIISTRING):
-                result += f": str"
-            elif bdt in (DataType.A_BYTEFIELD,):
-                result += f": bytes"
+                of.write(f": str")
+            elif bdt == DataType.A_BYTEFIELD:
+                of.write(f": bytes")
             elif bdt.name.startswith("A_FLOAT"):
-                result += f": float"
+                of.write(f": float")
             elif bdt.name.startswith("A_UINT"):
-                result += f": uint"
+                of.write(f": uint")
             elif bdt.name.startswith("A_INT"):
-                result += f": int"
+                of.write(f": int")
             else:
-                result += f": <unknown type>"
+                of.write(f": <unknown type {{ bdt.name }}>")
 
-            if (bl := dop.get_static_bit_length()) is not None:
-                result += f"{bl}"
-
-            result += "\n"
+            of.write("\n")
 
         elif isinstance(cm, LinearCompuMethod):
-            result += f": float\n"
+            bdt = dop.physical_type.base_data_type
+            if bdt in (DataType.A_UTF8STRING, DataType.A_UNICODE2STRING, DataType.A_ASCIISTRING):
+                of.write(f": str")
+            elif bdt in (DataType.A_BYTEFIELD,):
+                of.write(f": bytes")
+            elif bdt.name.startswith("A_FLOAT"):
+                of.write(f": float")
+            elif bdt.name.startswith("A_UINT"):
+                of.write(f": uint")
+            elif bdt.name.startswith("A_INT"):
+                of.write(f": int")
+            else:
+                of.write(f": <unknown type>")
+
             ll = cm.physical_lower_limit
             ul = cm.physical_upper_limit
-            if ll is None:
-                ll_str = "(inf"
+            if ll is None or ll.interval_type == IntervalType.INFINITE:
+                ll_str = "(-inf"
             else:
-                ll_delim = '[' if ll.interval_type == IntervalType.CLOSED else '('
+                ll_delim = '(' if ll.interval_type == IntervalType.OPEN else '['
                 ll_str = f"{ll_delim}{ll._value!r}"
 
-            if ul is None:
+            if ul is None or ul.interval_type == IntervalType.INFINITE:
                 ul_str = "inf)"
             else:
-                ul_delim = ']' if ul.interval_type == IntervalType.CLOSED else ')'
+                ul_delim = ')' if ul.interval_type == IntervalType.OPEN else ']'
```

Comment:
this logic is better off being in the __str__ of the class

Reply:
Not in the limit class, but in its container

Reply:
you mean in `CompuMethod`? I'd rather leave its `__str__()` method auto-generated by dataclass. This way it is guaranteed to be complete...